### PR TITLE
HDDS-1842. Implement S3 Abort MPU request to use Cache and DoubleBuffer.

### PR DESCRIPTION
**Pull request by bharatviswa504**

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/s3/multipart/S3MultipartUploadAbortResponse.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  @Override
+  public void addToDBBatch(OMMetadataManager omMetadataManager,
+      BatchOperation batchOperation) throws IOException {
+
+    if (getOMResponse().getStatus() == OzoneManagerProtocolProtos.Status.OK) {
+
+      // Delete from openKey table and multipart info table.
+      omMetadataManager.getOpenKeyTable().deleteWithBatch(batchOperation,
+          multipartKey);
+      omMetadataManager.getMultipartInfoTable().deleteWithBatch(batchOperation,
```
Comment (arp7) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/s3/multipart/S3MultipartUploadAbortResponse.java]: Indentation looks wrong perhaps?

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 162 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 7 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 52 | Maven dependency ordering for branch |
| +1 | mvninstall | 847 | trunk passed |
| +1 | compile | 450 | trunk passed |
| +1 | checkstyle | 91 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 1182 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 212 | trunk passed |
| 0 | spotbugs | 515 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 761 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 34 | Maven dependency ordering for patch |
| +1 | mvninstall | 719 | the patch passed |
| +1 | compile | 460 | the patch passed |
| +1 | javac | 460 | the patch passed |
| +1 | checkstyle | 99 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 880 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 217 | the patch passed |
| +1 | findbugs | 851 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 367 | hadoop-hdds in the patch passed. |
| -1 | unit | 2810 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 45 | The patch does not generate ASF License warnings. |
| | | 10449 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.Test2WayCommitInRatis |
|   | hadoop.ozone.client.rpc.TestFailureHandlingByClient |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.ozone.client.rpc.TestContainerStateMachine |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.TestStorageContainerManager |
|   | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.0 Server=19.03.0 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1155/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1155 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 595b18f9838e 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 1d98a21 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1155/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1155/1/testReport/ |
| Max. process+thread count | 3591 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1155/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by bharatviswa504** [comment]
Thank You @arp7 for the review.
Test failures are not related to this patch. I will commit this to the trunk.

**Pull request merged**